File: src/memory/schema.ts
export type MemoryType = "diary" | "fact" | "event" | "schedule" | "generic";
export type MemoryItem = {
  memory_id: string;
  created_at: string;
};

export type MemoryState = {
  title?: string
  memory_type?: MemoryType
  content?: unknown
}

export type MemoryNode = {
  node_id: string
  memory_id: string
  parent_node_id?: string
  created_at: string
  payload: Partial<MemoryState>
  change_reason?: string
}

File: src-tauri/src/schema.rs
use chrono::{DateTime, Utc};
pub struct MemoryItem {
    pub memory_id: String,
    pub created_at: DateTime<Utc>,
    pub memory_type: MemoryType,
    pub title: String,
}
pub enum MemoryType {
    Diary,
    Fact,
    Event,
    Schedule,
    Generic
}

pub struct MemoryNode {
    pub node_id: String,
    pub memory_id: String,
    pub parent_node_id: Option<String>,
    pub created_at: DateTime<Utc>,
    pub content: serde_json::Value,
}
